test/unit/alphabet/custom_alphabet4_test.cpp

```diff
@@ -0,0 +1,36 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+#include <iostream>
+
+#include <seqan3/alphabet/concept.hpp>
+
+// see issue https://github.com/seqan/seqan3/issues/1518
+
+struct alphabet2_shim
+{
+    using rank_t = uint8_t;
+    using char_t = char;
+
+    rank_t to_rank() const noexcept;
+
+    char_t to_char() const noexcept;
+
+    constexpr static bool alphabet_size{1};
+
+    friend bool operator<(alphabet2_shim, alphabet2_shim);
+    friend bool operator<=(alphabet2_shim, alphabet2_shim);
+    friend bool operator>(alphabet2_shim, alphabet2_shim);
+    friend bool operator>=(alphabet2_shim, alphabet2_shim);
+    friend bool operator==(alphabet2_shim, alphabet2_shim);
+    friend bool operator!=(alphabet2_shim, alphabet2_shim);
+};
+
+static_assert(seqan3::semialphabet<alphabet2_shim>);
+static_assert(!seqan3::writable_semialphabet<alphabet2_shim>);
+static_assert(seqan3::alphabet<alphabet2_shim>);
+static_assert(!seqan3::writable_alphabet<alphabet2_shim>);
```

Comment:
Can we make this a proper google test?

Reply:
Sure. I just wanted to push early because hannes was active and I saw my chance to get an explanation :D